### DISCUSSION TITLE
How to add multiple routes to the multi depot VRP?

### DISCUSSION THREAD

**Discussion by KatarinaDajic** [PyVRP/PyVRP#574] [Q&A] [answered]
Hi guys, on the link https://github.com/PyVRP/PyVRP/issues/366, I found that you have started some discussions regarding multi-trip VRP, in which vehicles can return to the depot after serving one route, and then serve customers from another route. In my problem vehicles have a capacity and they have to serve a large number of customers, for which the vehicle capacity is not sufficient. For each depot that I have, multiple routes should be created, as the problem is based on a real-life large scale instances. I have read your thread, but I am still far away from some concrete idea, as I am just a beginner in the coding world. Do you have an idea now how to tackle it, or maybe if it is planned as a new feature for the next release?

Thank you!!:)

**Comment by N-Wouda** [comment]
Hi @KatarinaDajic! 👋 

In order:

> [...] is planned as a new feature for the next release?

Multi-trip is indeed not something we support yet. It'll probably be supported at some point in the future, but there are no concrete plans for this right now. It is not likely to be part of the next release, so the earliest point at which we could support this is after the summer.

> Do you have an idea now how to tackle it

Not any more than what's been summarised in #366. Does the content of [this comment](https://github.com/PyVRP/PyVRP/issues/366#issuecomment-1904056219) help you get started? I haven't given this much more thought than what's available there, so that should be your best bet for now. You might also want to read a few papers linked in #366 as background reading.

In any case, multi-trip is a non-trivial addition that may not be so easy to implement. If you're unfamiliar with C++ and/or Python I would probably not attempt it, but instead try to find a workaround that works reasonably well for your practical use case. Would, for example, duplicating a vehicle into multiple, non-overlapping shifts be a good enough approximation?

**Reply by N-Wouda** [reply, marked as answer]
#366 / multi-trip will be part of the upcoming release (0.11).

**Reply by N-Wouda** [reply]
PyVRP 0.11 (with multi-trip support) has been released a few days ago. So this is now generally available.
